openglyph/src/io/mega_filesystem.cpp

```diff
@@ -0,0 +1,69 @@
+#include <khepri/log/log.hpp>
+#include <khepri/utility/string.hpp>
+
+#include <openglyph/io/mega_file.hpp>
+#include <openglyph/io/mega_filesystem.hpp>
+#include <openglyph/parser/xml_parser.hpp>
+
+namespace openglyph::io {
+const khepri::log::Logger LOG("megafs");
+
+MegaFileSystem::MegaFileSystem(std::vector<std::filesystem::path> data_paths)
+    : m_data_paths(std::move(data_paths))
+{
+    for (const auto& data_path : m_data_paths) {
+        const std::filesystem::path index_file = data_path / "Data/megafiles.xml";
```

Comment:
Write this as `data_path / "Data" / "megafiles.xml"` to keep consistent folder separators.